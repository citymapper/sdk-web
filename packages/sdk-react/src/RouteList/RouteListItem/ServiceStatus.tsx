import { Status } from '@citymapper/api/@types'
import { cx } from '@emotion/css'
import React, { useMemo } from 'react'
import IconStatusDisrupted from '../../Icons/IconStatusDisrupted'
import IconStatusPrevented from '../../Icons/IconStatusPrevented'
import styled from '../../utils/cmStyled'

interface ServiceStatusProps {
  type: Status['type'] | null
  isBox?: boolean
  multiService?: boolean
}

const Wrapper = styled('div', {
  className: 'cm-route-summary-service-status',
})<{ small?: boolean }>`
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;

  svg {
    width: 100%;
    height: 100%;
  }

  &.cm-route-summary-service-status {
    &--box {
      top: -8px;
      right: -8px;
    }

    &--small {
      width: 11px;
      height: 11px;
      top: -6px;
      right: -4px;
    }
  }
`

const ServiceStatus: React.FC<ServiceStatusProps> = ({
  type,
  isBox,
  multiService,
}) => {
  const disruptionIcon = useMemo(() => {
    switch (type) {
      case 'travel_affected':
        return <IconStatusDisrupted />
      case 'travel_prevented':
        return <IconStatusPrevented />
      default:
        return null
    }
  }, [type])

  if (!disruptionIcon) return null

  return (
    <Wrapper
      className={cx(
        {
          'cm-route-summary-service-status--small': multiService,
          'cm-route-summary-service-status--box': isBox,
        },
        `cm-route-summary-service-status--${type}`
      )}
    >
      {disruptionIcon}
    </Wrapper>
  )
}

export default ServiceStatus
