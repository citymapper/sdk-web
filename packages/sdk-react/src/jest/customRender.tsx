import { render, RenderOptions } from '@testing-library/react'
import React, { JSXElementConstructor, ReactElement } from 'react'
import CMProvider from '../CMProvider'

const TestWrapper: React.FC<{ children: React.ReactElement }> = ({
  children,
}) => (
  <CMProvider
    config={{
      apiKey: '',
      citymapperFont: false,
    }}
  >
    {children}
  </CMProvider>
)

const customRender = (
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import('@testing-library/dom/types/queries'),
    HTMLElement,
    HTMLElement
  >
) => render(ui, { wrapper: TestWrapper, ...options })

export default customRender
