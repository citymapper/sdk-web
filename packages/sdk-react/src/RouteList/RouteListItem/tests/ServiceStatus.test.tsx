import React from 'react'
import customRender from '../../../jest/customRender'
import ServiceStatus from '../ServiceStatus'

describe('ServiceStatus', () => {
  it('travel_affected should render correctly with default props', () => {
    const { container } = customRender(<ServiceStatus type="travel_affected" />)
    expect(container).toMatchSnapshot()
  })

  it('travel_affected should render correctly with isBox prop', () => {
    const { container } = customRender(
      <ServiceStatus type="travel_affected" isBox />
    )
    expect(container).toMatchSnapshot()
  })

  it('travel_affected should render correctly with multiService prop', () => {
    const { container } = customRender(
      <ServiceStatus type="travel_affected" multiService />
    )
    expect(container).toMatchSnapshot()
  })

  it('travel_prevented should render correctly with default props', () => {
    const { container } = customRender(
      <ServiceStatus type="travel_prevented" />
    )
    expect(container).toMatchSnapshot()
  })

  it('travel_prevented should render correctly with isBox prop', () => {
    const { container } = customRender(
      <ServiceStatus type="travel_prevented" isBox />
    )
    expect(container).toMatchSnapshot()
  })

  it('travel_prevented should render correctly with multiService prop', () => {
    const { container } = customRender(
      <ServiceStatus type="travel_prevented" multiService />
    )
    expect(container).toMatchSnapshot()
  })
})
