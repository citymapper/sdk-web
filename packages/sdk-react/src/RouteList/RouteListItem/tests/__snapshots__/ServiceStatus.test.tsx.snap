// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceStatus travel_affected should render correctly with default props 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--travel_affected emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#FE9001"
        r="7.5"
        stroke="white"
      />
      <path
        d="M7.99979 5.97512C8.68949 5.97512 9.2486 5.42105 9.2486 4.73756C9.2486 4.05407 8.68949 3.5 7.99979 3.5C7.31009 3.5 6.75098 4.05407 6.75098 4.73756C6.75098 5.42105 7.31009 5.97512 7.99979 5.97512Z"
        fill="white"
      />
      <path
        d="M7.9998 7.25414C7.44804 7.25414 7.00075 7.70143 7.00075 8.25319V11.001C7.00075 11.5527 7.44804 12 7.9998 12C8.55156 12 8.99884 11.5527 8.99884 11.001V8.25319C8.99884 7.70143 8.55156 7.25414 7.9998 7.25414Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`ServiceStatus travel_affected should render correctly with isBox prop 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--box cm-route-summary-service-status--travel_affected emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#FE9001"
        r="7.5"
        stroke="white"
      />
      <path
        d="M7.99979 5.97512C8.68949 5.97512 9.2486 5.42105 9.2486 4.73756C9.2486 4.05407 8.68949 3.5 7.99979 3.5C7.31009 3.5 6.75098 4.05407 6.75098 4.73756C6.75098 5.42105 7.31009 5.97512 7.99979 5.97512Z"
        fill="white"
      />
      <path
        d="M7.9998 7.25414C7.44804 7.25414 7.00075 7.70143 7.00075 8.25319V11.001C7.00075 11.5527 7.44804 12 7.9998 12C8.55156 12 8.99884 11.5527 8.99884 11.001V8.25319C8.99884 7.70143 8.55156 7.25414 7.9998 7.25414Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`ServiceStatus travel_affected should render correctly with multiService prop 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--small cm-route-summary-service-status--travel_affected emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#FE9001"
        r="7.5"
        stroke="white"
      />
      <path
        d="M7.99979 5.97512C8.68949 5.97512 9.2486 5.42105 9.2486 4.73756C9.2486 4.05407 8.68949 3.5 7.99979 3.5C7.31009 3.5 6.75098 4.05407 6.75098 4.73756C6.75098 5.42105 7.31009 5.97512 7.99979 5.97512Z"
        fill="white"
      />
      <path
        d="M7.9998 7.25414C7.44804 7.25414 7.00075 7.70143 7.00075 8.25319V11.001C7.00075 11.5527 7.44804 12 7.9998 12C8.55156 12 8.99884 11.5527 8.99884 11.001V8.25319C8.99884 7.70143 8.55156 7.25414 7.9998 7.25414Z"
        fill="white"
      />
    </svg>
  </div>
</div>
`;

exports[`ServiceStatus travel_prevented should render correctly with default props 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--travel_prevented emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#D22020"
        r="7.5"
        stroke="white"
      />
      <path
        clip-rule="evenodd"
        d="M4.79289 4.79289C5.18342 4.40237 5.81658 4.40237 6.20711 4.79289L8 6.58579L9.79289 4.79289C10.1834 4.40237 10.8166 4.40237 11.2071 4.79289C11.5976 5.18342 11.5976 5.81658 11.2071 6.20711L9.41421 8L11.2071 9.79289C11.5976 10.1834 11.5976 10.8166 11.2071 11.2071C10.8166 11.5976 10.1834 11.5976 9.79289 11.2071L8 9.41421L6.20711 11.2071C5.81658 11.5976 5.18342 11.5976 4.79289 11.2071C4.40237 10.8166 4.40237 10.1834 4.79289 9.79289L6.58579 8L4.79289 6.20711C4.40237 5.81658 4.40237 5.18342 4.79289 4.79289Z"
        fill="white"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`ServiceStatus travel_prevented should render correctly with isBox prop 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--box cm-route-summary-service-status--travel_prevented emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#D22020"
        r="7.5"
        stroke="white"
      />
      <path
        clip-rule="evenodd"
        d="M4.79289 4.79289C5.18342 4.40237 5.81658 4.40237 6.20711 4.79289L8 6.58579L9.79289 4.79289C10.1834 4.40237 10.8166 4.40237 11.2071 4.79289C11.5976 5.18342 11.5976 5.81658 11.2071 6.20711L9.41421 8L11.2071 9.79289C11.5976 10.1834 11.5976 10.8166 11.2071 11.2071C10.8166 11.5976 10.1834 11.5976 9.79289 11.2071L8 9.41421L6.20711 11.2071C5.81658 11.5976 5.18342 11.5976 4.79289 11.2071C4.40237 10.8166 4.40237 10.1834 4.79289 9.79289L6.58579 8L4.79289 6.20711C4.40237 5.81658 4.40237 5.18342 4.79289 4.79289Z"
        fill="white"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`ServiceStatus travel_prevented should render correctly with multiService prop 1`] = `
.emotion-0 {
  position: absolute;
  width: 16px;
  height: 16px;
  top: -8px;
  right: -8px;
  z-index: 100;
}

.emotion-0 svg {
  width: 100%;
  height: 100%;
}

.emotion-0.cm-route-summary-service-status--box {
  top: -8px;
  right: -8px;
}

.emotion-0.cm-route-summary-service-status--small {
  width: 11px;
  height: 11px;
  top: -6px;
  right: -4px;
}

.emotion-1 {
  width: 24px;
  height: 24px;
  width: 16;
  height: 16;
}

<div>
  <div
    class="cm-route-summary-service-status cm-route-summary-service-status--small cm-route-summary-service-status--travel_prevented emotion-0"
  >
    <svg
      class="emotion-1"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
    >
      <circle
        cx="8"
        cy="8"
        fill="#D22020"
        r="7.5"
        stroke="white"
      />
      <path
        clip-rule="evenodd"
        d="M4.79289 4.79289C5.18342 4.40237 5.81658 4.40237 6.20711 4.79289L8 6.58579L9.79289 4.79289C10.1834 4.40237 10.8166 4.40237 11.2071 4.79289C11.5976 5.18342 11.5976 5.81658 11.2071 6.20711L9.41421 8L11.2071 9.79289C11.5976 10.1834 11.5976 10.8166 11.2071 11.2071C10.8166 11.5976 10.1834 11.5976 9.79289 11.2071L8 9.41421L6.20711 11.2071C5.81658 11.5976 5.18342 11.5976 4.79289 11.2071C4.40237 10.8166 4.40237 10.1834 4.79289 9.79289L6.58579 8L4.79289 6.20711C4.40237 5.81658 4.40237 5.18342 4.79289 4.79289Z"
        fill="white"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
